### PR TITLE
Type fixes for DDC.

### PR DESCRIPTION
**Pull request by davidmorgan**
This change is [Reviewable](https://reviewable.io/reviews/google/built_value.dart/355)

### CODE REVIEW

**Comment by davidmorgan** [comment]
(Travis fails because the versions aren't released--the tests pass locally--I will make sure travis also passes locally when releasing. To do this properly I need to fix #354)

File: built_value/lib/src/built_set_multimap_serializer.dart [hunk 1]
```diff
@@ -62,11 +62,10 @@ class BuiltSetMultimapSerializer
     for (int i = 0; i != serialized.length; i += 2) {
       final key = serializers.deserialize(serialized.elementAt(i),
           specifiedType: keyType);
-      result.addValues(
-          key,
-          serialized.elementAt(i + 1).map((value) =>
-                  serializers.deserialize(value, specifiedType: valueType))
-              as Iterable);
+      for (final value in serialized.elementAt(i + 1).map((value) =>
+          serializers.deserialize(value, specifiedType: valueType))) {
```
Comment (manuelF) [inline built_value/lib/src/built_set_multimap_serializer.dart]: I'd extract this function out of the for,  too complex.

File: built_value/lib/src/built_list_multimap_serializer.dart [hunk 2]
```diff
@@ -64,11 +64,10 @@ class BuiltListMultimapSerializer
     for (int i = 0; i != serialized.length; i += 2) {
       final key = serializers.deserialize(serialized.elementAt(i),
           specifiedType: keyType);
-      result.addValues(
-          key,
-          serialized.elementAt(i + 1).map((value) =>
-                  serializers.deserialize(value, specifiedType: valueType))
-              as Iterable);
+      for (final value in serialized.elementAt(i + 1).map((value) =>
+          serializers.deserialize(value, specifiedType: valueType))) {
```
Comment (manuelF) [inline built_value/lib/src/built_list_multimap_serializer.dart]: extract too

File: built_value/lib/src/built_set_multimap_serializer.dart (hunk 1 above)
Comment (davidmorgan) [reply built_value/lib/src/built_set_multimap_serializer.dart]: Done.

File: built_value/lib/src/built_list_multimap_serializer.dart (hunk 2 above)
Comment (davidmorgan) [reply built_value/lib/src/built_list_multimap_serializer.dart]: Done. Thanks.

**Pull request merged**